src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -115,6 +116,26 @@ public ConnectionService()
             var defaultQueue = new ConnectedBindingQueue(needsMetadata: false);
             connectedQueues.AddOrUpdate("Default", defaultQueue, (key, old) => defaultQueue);
             this.LockedDatabaseManager.ConnectionService = this;
+            
+            SqlColumnEncryptionAzureKeyVaultProvider sqlColumnEncryptionAzureKeyVaultProvider = new SqlColumnEncryptionAzureKeyVaultProvider(AzureActiveDirectoryAuthenticationCallback);
+            SqlConnection.RegisterColumnEncryptionKeyStoreProviders(customProviders: new Dictionary<string, SqlColumnEncryptionKeyStoreProvider>(capacity: 1, comparer: StringComparer.OrdinalIgnoreCase)

```

Comment:
This isn't going to work with how the tests are currently set up to Mock the connection service which will create multiple instances of this. You can probably get around that by putting this in a static constructor instead.

Reply:
Creating a static constructor for this code chunk should work if I can make this member property static. Is this okay or do you think that this will this cause problems too? 
``` csharp
internal static IProtocolEndpoint ServiceHost { get; set; }
```
The reason is because, while I am instantiating and setting the KeyVaultProvider object on the SqlConnection statically, the callback function that requests the token from ADS depends on the ServiceHost ProtocolEndpoint which is not static.

Reply:
Looking at it more, A better option may be to simply grab a singleton instance of ServiceHost from within the callback function and use that rather than the ConnectionService's ServiceHost member property. I'll do that instead.

Reply:
Fixed.